### PR TITLE
#67 Star size residual correlation functions

### PR DESCRIPTION
**Pull request by msimet**
And here's the star size residual correlation function: a scalar correlation function of `(psf_sigma - sigma)/psf_sigma`.  Pretty straightforward, except for revealing a couple of bugs in pieces of code that we haven't really touched on before...

### CODE REVIEW

File: stile/hsc/base_tasks.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ class CCDSingleEpochStileConfig(lsst.pex.config.Config):
     sys_tests = adapter_registry.makeField("tests to run", multi=True,
                     default=[#"StatsPSFFlux", #"GalaxyXGalaxyShear", "BrightStarShear",
                              "StarXGalaxyShear", "StarXStarShear", "Rho1",
+                             "StarSizeResidual",
```
Comment (HironaoMiyatake) [inline stile/hsc/base_tasks.py]: It's not clear that it is a correlation function from this name. Maybe `StarXStarSizeResidual` is better?

**Comment by HironaoMiyatake** [comment]
I went through the code, and wrote one comment about the short name of this test. I found a bug; `data` was put into a method to calculate a correlation function, but it should be `new_data`.

After @msimet  check out those, I think this branch can be merged into master.

**Comment by msimet** [comment]
I changed the name of the test, as suggested.

I also made one other change (to make sure the sigma columns are calculated for data sets other than the first one, if they're present).  Do you want to look at the code to make sure it looks okay to you?  The commit history looks a little weird, unfortunately--my text editor didn't reload the file when I thought it did, so I ended up changing some things back that I didn't mean to...

**Comment by HironaoMiyatake** [comment]
I think it would be better to change the denominator of 
`new_data['sigma'] = (new_data['psf_sigma'] - new_data['sigma'])/new_data['sigma']`
to `new_data['psf_sigma']`, to make it less noiser (this is what I do for other fractional residuals).

Also you use 
`objects_list = ['star']`
, but do not we want to use 
`objects_list = ['star PSF']`
? At least `rho1` is the latter, so it would be nice to make it consistent.

**Comment by msimet** [comment]
Okay, made both these changes--does that look good?

**Comment by HironaoMiyatake** [comment]
Yes, I think it's ready to merge!

**Comment by rmandelb** [comment]
(I realize you don't want to put that on here since it's proprietary data, but could you e-mail them?)

**Comment by msimet** [comment]
I tested this on a few little patches but haven't run the whole of the old (or new) reductions.  I can certainly do that by meeting next week, though, assuming Coma gets up and running...

**Comment by msimet** [comment]
(I'm gonna merge this, but we can keep talking on the closed issue if we need to)

**Pull request merged**